src/transformers/models/whisper/modeling_whisper.py

```diff
@@ -1533,6 +1609,10 @@ def generate(
                 provided as a prompt to each chunk. This can be used to provide or "prompt-engineer" a context for
                 transcription, e.g. custom vocabularies or proper nouns to make it more likely to predict those words
                 correctly. It cannot be used in conjunction with `decoder_start_token_id` as it overwrites this value.
+            return_token_timestamps (`bool`, *optional*):
+                Whether to return token-level timestamps with the text. This can be used with or without the
+                `return_timestamps` option. To get word-level timestamps, use the tokenizer to group the tokens into
+                words.
```

Comment:
Might be worth noting here that this will be slow if the length of the sequences are big? The code is not vectorized at all, but I haven't tried to run it so not sure how slow it actually is.

Reply:
I don't think it's particularly slow. Most of the code involved _is_ vectorized, only the dynamic time warping isn't. That does slow down with the length of the sequences but running the actual model is way slower, so I don't think it matters.